src/services/authenticator.js

```diff
@@ -71,6 +74,77 @@ function Authenticator() {
       });
     });
   };
+
+  this.loginWithEmailOrTokenArgv = async (config) => {
+    const token = await this.login(config);
+    this.saveToken(token);
+    return null;
+  };
+
+  this.loginWithToken = (token) => {
+    const validationResult = this.validateToken(token);
+    if (validationResult !== true) {
+      throw new Error(validationResult);
+    }
+    return token;
+  };
+
+  this.login = async ({ email, password, token }) => {
+    if (!email) email = await this.promptEmail();
+
+    if (token) {
+      return this.loginWithToken(token);
+    }
+
+    const isGoogleAccount = await api.isGoogleAccount(email);
+    if (isGoogleAccount) {
+      return this.loginWithGoogle();
+    }
+
+    return this.loginWithPassword(email, password);
+  };
+
+  this.promptEmail = async () => {
+    const { email } = await inquirer.prompt([{
+      type: 'input',
+      name: 'email',
+      message: 'What\'s your email address?',
+      validate: (input) => {
+        if (EMAIL_REGEX.test(input)) {
+          return true;
+        }
+        return input ? 'Invalid email' : 'Please enter your email address.';
+      },
+    }]);
+    return email;
+  };
+
+  this.loginWithGoogle = async () => {
+    const endpoint = process.env.FOREST_URL && process.env.FOREST_URL.includes('localhost')
+      ? 'http://localhost:4200' : 'https://app.forestadmin.com';
+    const url = chalk.cyan.underline(`${endpoint}/authentication-token`);
+    logger.info(`To authenticate with your Google account, please follow this link and copy the authentication token: ${url}`);
+
+    const { sessionToken } = await inquirer.prompt([{
+      type: 'password',
+      name: 'sessionToken',
+      message: 'Enter your Forest Admin authentication token:',
+      validate: this.validateToken,
+    }]);
+    return sessionToken;
+  };
+
+  this.loginWithPassword = async (email, password) => {
+    if (!password) {
+      ({ password } = await inquirer.prompt([{
+        type: 'password',
+        name: 'password',
+        message: 'What\'s your Forest Admin password:',
```

Comment:
Just a suggestion

```suggestion
        message: 'What is your Forest Admin password:',
```

Or (to be consistant with token prompt):

```suggestion
        message: 'Enter your Forest Admin password:',
```